blog-posts/staticIpLambda/staticIpLambda.md

```diff
@@ -0,0 +1,193 @@
+
+First, we need to create a VPC with a public subnet and a private subnet. The private subnet will host our lambda function and the public subnet will host our NAT Gateway. <br>
+
+```ts
+const vpc = new Vpc(stack, 'Vpc', {
+	vpcName: 'vpc',
+	natGateways: 1, // 👈 Automatically creates an Elastic IP
+	maxAzs: 1, // 👈 Use more if you need high availability
+	subnetConfiguration: [
+		{
+		name: 'private-subnet-1',
+		subnetType: SubnetType.PRIVATE_WITH_NAT,
+		cidrMask: 24,
+		},
+		{
+		name: 'public-subnet-1',
+		subnetType: SubnetType.PUBLIC,
+		cidrMask: 24,
+		},
+	],
+});
+```
+> 💸 **Pricing Disclaimer**
+> This architecture will cost **some money**. The Nat Gateway runs on a EC2 instance which will cost you **around 30$/month**. This price is multiplied by the number of AZs you use. 
+ 
+ I also recommend that you use CloudFormation Outputs to store the IDs of the resources you will need later on. We want to slap that *✨clean IaC code✨* in our PR 🫱.
+ 
+ ```ts
+
+export const vpcSecurityGroupOutputId = 'SgOutputId';
+export const vpcPrivateSubnetOutputId = 'VpcPvSubOutputId';
+...
+
+const privateSubnets = vpc.selectSubnets(
+	{ subnetType: SubnetType.PRIVATE_WITH_NAT }
+);
+const [privateSubnetId1] = privateSubnets.subnetIds;
+
+new CfnOutput(stack, vpcSecurityGroupOutputId, {
+	value: vpc.vpcDefaultSecurityGroup,
+});
+new CfnOutput(stack, vpcPrivateSubnet1OutputId, {
+	value: privateSubnetId1,
+});
+
+```
+
+## ⚡ Deploy the Lambda function
+
+Now that we have our VPC, we can deploy our lambda function, using the Serverless Framework. 
+
+```ts
+
+// A small util to get CF template outputs 👇
+const getOutputValue = (
+	template: CloudFormationTemplate,
+	outputId: string
+ ): string => template.Outputs?.[outputId];
+
+export const staticIpLambda = {
+  timeout: 15,
+  handler: getHandlerPath(__dirname),
+  vpc: {
+    securityGroupIds: [
+		getOutputValue(
+		 cdkAppCloudFormationTemplate,
+		 vpcSecurityGroupOutputId
+		 ),
+	],
+    subnetIds: [
+		getOutputValue(
+		  cdkAppCloudFormationTemplate,
+		  vpcPrivateSubnet1OutputId
+		  ),
+		],
+  	},
+};
+```
+
+And bam! 🎉 Our lambda function is deployed in our private subnet. <br>
+All the outbound traffic from our lambda function will now go through the NAT Gateway, which will use the Elastic IP we created earlier. 🚀
+
+---
+
+You can retrieve the Elastic IP of your NAT Gateway using the AWS Console.
+Go to the VPC service and click Elastic IPs. You should see the Elastic IP created by the CDK. <br>
+
+![Elastic IP](./assets/elasticIp.png 'Elastic IP')
+
+Now just message your partner and ask them to whitelist this IP. 📩
+
+# 2️⃣ Perform SFTP operations in your Lambda function
+
+Now let's talk about a very specific use case: **performing SFTP operations in your lambda function**. <br>
+I wanted to share with you this use case because I came across weird issues with the `ssh2-promise` package not being correctly bundled into my lambda `.zip code`. 🤯
```

Comment:
Move this to the introduction of your article? Add that is is related to serverless-esbuild. Say that this part is actually optional, so that people with a different use case but who need whitelisting might still be interested in the first part :)
Maybe worth writing two articles?

Reply:
j'ai rajouté ces infos dans un petit disclaimer ! par contre je pense pas qu'il y ait assez de contenu pour deux articles 🙈